be/log_discard.c

```diff
@@ -230,72 +227,73 @@ static void be_log_discard_item_discard_ast(struct m0_sm_group *grp,
 {
 	struct m0_be_log_discard_item *ldi = ast->sa_datum;
 	struct m0_be_log_discard      *ld  = ldi->ldi_ld;
+	struct m0_be_log_discard_item tmp_ldi;
+	struct m0_ext *ext;
 
-	M0_ENTRY("ld=%p ldi=%p", ld, ldi);
-
-	ld->lds_cfg.ldsc_discard(ld, ldi);
-
+	/* if lds_sync_item is NULL all the ldi are already discarded
+	   and lds_start_q will not have any ldi with LDI_FINISHED state */
+	if (ld->lds_sync_item == NULL)
+		return;
```

Comment:
[* defect *] It wouldn't be possible to catch log discard bugs if some asts would be silently discarded.

Reply:
Working on solution to post an ast after fdatasync() (if there is no ast running already) and discard each log record sequentially in a loop in that ast.